statistics/histogram.go

```diff
@@ -321,6 +323,29 @@ func (hg *Histogram) BucketToString(bktID, idxCols int) string {
 	return fmt.Sprintf("num: %d lower_bound: %s upper_bound: %s repeats: %d", hg.bucketCount(bktID), lowerVal, upperVal, hg.Buckets[bktID].Repeat)
 }
 
+// RemoveIdxVals remove the given values from the histogram.
+func (hg *Histogram) RemoveIdxVals(idxValCntPairs []TopNMeta) {
+	sort.Slice(idxValCntPairs, func(i, j int) bool {
+		return bytes.Compare(idxValCntPairs[i].Encoded, idxValCntPairs[j].Encoded) < 0
+	})
+	totalSubCnt := int64(0)
+	for bktIdx, pairIdx := 0, 0; bktIdx < hg.Len(); bktIdx++ {
+		for pairIdx < len(idxValCntPairs) {
+			cmpResult := bytes.Compare(hg.Bounds.Column(0).GetBytes(bktIdx*2+1), idxValCntPairs[pairIdx].Encoded)
+			if cmpResult < 0 {
+				break
+			}
+			totalSubCnt += int64(idxValCntPairs[pairIdx].Count)
```

Comment:
I think we missed the case in which an element in topn might fall in the gap between two buckets.